File: gpu_bartender/finetuning_args.py
from dataclasses import dataclass, field
from typing import Optional

@dataclass
class LoraArgs:
    lora_alpha: Optional[int] = field(default=None)
    lora_dropout: Optional[float] = field(default=None)
    lora_rank: Optional[int] = field(default=8)
    lora_target: Optional[str] = field(default=None)

@dataclass
class QLoraArgs:
    qlora_alpha: Optional[int] = field(default=None)
    qlora_dropout: Optional[float] = field(default=None)

@dataclass
class FinetuningArgs(LoraArgs, QLoraArgs):
    training_precision: str = field(default='mixed')
    is_fsdp: bool = field(default=True)

File: gpu_bartender/src/finetuningArgs.ts
import { LoraArgs, QLoraArgs } from './loraArgs';

export class FinetuningArgs {
    public loraArgs: LoraArgs;
    public qloraArgs: QLoraArgs;

    constructor(
        public trainingPrecision: string = 'mixed',
        public isFsdp: boolean = true,
        loraAlpha?: number,
        loraDropout?: number,
        loraRank?: number,
        loraTarget?: string,
        qloraAlpha?: number,
        qloraDropout?: number
    ) {
        this.loraArgs = new LoraArgs(loraAlpha, loraDropout, loraRank, loraTarget);
        this.qloraArgs = new QLoraArgs(qloraAlpha, qloraDropout);
    }
}
